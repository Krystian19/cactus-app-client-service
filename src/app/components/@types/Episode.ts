import Anime from './Anime';
import Release from './Release';

export type Episode = {
  id: Number,
  thumbnail?: String,
  episodeOrder?: Number,
  EarlierEpisode?: Episode,
  LaterEpisode?: Episode,
  Release?: Release,
  Anime?: Anime,
};

export default Episode;
