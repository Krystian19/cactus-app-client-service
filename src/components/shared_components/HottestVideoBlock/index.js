import React, { Component } from 'react';
import {
  Link,
} from 'react-router-dom';
import LazyImage from '../LazyImage';

export default class HottestVideoBlock extends Component {
  // constructor(props) {
  //   super(props);
  // }

  render() {
    const {
      props: {
        episodes,
        title,
        history,
        viewAllLink,
      },
    } = this.props;

    return (
      <div className="video-block">
        <div className="video-block-title">
          <div className="video-block-title-head">
            <span>{title}</span>
          </div>
          {viewAllLink && (
            // <div className="video-block-title-options">
            //   <a href={viewAllLink} className="button">View all</a>
            // </div>
            <Link to={viewAllLink} className="button" tabIndex="-1">
              View all
            </Link>
          )}
        </div>
        <div className="video-block-content">
          <div className="anime-small-thumbnail-list">
            {episodes.map((episode, index) => (
              <div
                key={episode.id}
                className={`anime-small-thumbnail ${(index === 0) ? 'big' : 'small'}`}
                onClick={() => history.push(`/anime/video/${episode.id}`)}
                onKeyPress={() => history.push(`/anime/video/${episode.id}`)}
                role="menuitem"
                tabIndex={index}
              >
                <div className="cover">
                  <LazyImage
                    src={`/img_cdn/${episode.Season.background}`}
                    errorSrc="/img_cdn/test.jpg"
                    alt="thumbnail"
                    className="anime-small-thumbnail fade-in"
                  />
                  {/* <img
                    src={`/img_cdn/${episode.Season.background}`}
                    alt="thumbnail"
                    className="anime-small-thumbnail"
                  /> */}
                  <div
                    className="overlay hover_hidden darken"
                  >
                    <svg className="play" viewBox="0 0 24 24">
                      <path d="M8,5.14V19.14L19,12.14L8,5.14Z" />
                    </svg>
                  </div>
                </div>
                <div className="info">
                  <div className="title">
                    <div
                      className="title-container"
                    >
                      {`${episode.Season.title} (Season ${episode.Season.seasonOrder})`}
                    </div>
                    <div className="detail-container">
                      {`Ep. ${episode.episodeOrder}`}
                    </div>
                  </div>
                </div>
              </div>
            ))}
          </div>
        </div>
      </div>
    );
  }
}

// ({ Episodes }) => (
// const HottestVideoBlock = (props) => {
//   const {
//     props: {
//       episodes,
//       title,
//       history,
//       viewAllLink,
//     },
//   } = props;

//   return (
//     <div className="video-block">
//       <div className="video-block-title">
//         <div className="video-block-title-head">
//           <span>{title}</span>
//         </div>
//         {viewAllLink && (
//           // <div className="video-block-title-options">
//           //   <a href={viewAllLink} className="button">View all</a>
//           // </div>
//           <Link to={viewAllLink} className="button" tabIndex="-1">
//             View all
//           </Link>
//         )}
//       </div>
//       <div className="video-block-content">
//         <div className="anime-small-thumbnail-list">
//           {episodes.map((episode, index) => (
//             <div
//               key={episode.id}
//               className={`anime-small-thumbnail ${(index === 0) ? 'big' : 'small'}`}
//               onClick={() => history.push(`/anime/video/${episode.id}`)}
//               onKeyPress={() => history.push(`/anime/video/${episode.id}`)}
//               role="menuitem"
//               tabIndex={index}
//             >
//               <div className="cover">
//                 <LazyImage
//                   src={`/img_cdn/${episode.Season.background}`}
//                   errorSrc="/img_cdn/test.jpg"
//                   alt="thumbnail"
//                   className="anime-small-thumbnail fade-in"
//                 />
//                 {/* <img
//                   src={`/img_cdn/${episode.Season.background}`}
//                   alt="thumbnail"
//                   className="anime-small-thumbnail"
//                 /> */}
//                 <div
//                   className="overlay hover_hidden darken"
//                 >
//                   <svg className="play" viewBox="0 0 24 24">
//                     <path d="M8,5.14V19.14L19,12.14L8,5.14Z" />
//                   </svg>
//                 </div>
//               </div>
//               <div className="info">
//                 <div className="title">
//                   <div
//                     className="title-container"
//                   >
//                     {`${episode.Season.title} (Season ${episode.Season.seasonOrder})`}
//                   </div>
//                   <div className="detail-container">
//                     {`Ep. ${episode.episodeOrder}`}
//                   </div>
//                 </div>
//               </div>
//             </div>
//           ))}
//         </div>
//       </div>
//     </div>
//   );
// };

// export default HottestVideoBlock;

// ðŸ”¥ right now

// <div className="anime-small-thumbnail big">
// <div className="cover">
//   <img src="https://cdn.masterani.me/wallpaper/0/28163fr0Ltd5.jpg" alt="thumbnail" className="anime-small-thumbnail" />
//   <div className="overlay hover_hidden darken">
//     <svg className="play" viewBox="0 0 24 24">
//       <path d="M8,5.14V19.14L19,12.14L8,5.14Z" />
//     </svg>
//   </div>
// </div>
// <div className="info">
//   <div className="title">
//     <div className="title-container"> Boku No hero Academia 3rd season </div>
//     <div className="detail-container"> Ep. 16 </div>
//   </div>
// </div>
// </div>
// <div className="anime-small-thumbnail small">
// <div className="cover">
//   <img src="https://cdn.masterani.me/wallpaper/0/64i2RVIJeV.jpg" alt="thumbnail" className="anime-small-thumbnail" />
//   <div className="overlay hover_hidden darken">
//     <svg className="play" viewBox="0 0 24 24">
//       <path d="M8,5.14V19.14L19,12.14L8,5.14Z" />
//     </svg>
//   </div>
// </div>
// <div className="info">
//   <div className="title">
//     <div className="title-container"> One Piece </div>
//     <div className="detail-container"> Ep. 266 </div>
//   </div>
// </div>
// </div>
// <div className="anime-small-thumbnail small">
// <div className="cover">
//   <img src="https://cdn.masterani.me/wallpaper/0/2909rbi6tQo0.jpg" alt="thumbnail" className="anime-small-thumbnail" />
//   <div className="overlay hover_hidden darken">
//     <svg className="play" viewBox="0 0 24 24">
//       <path d="M8,5.14V19.14L19,12.14L8,5.14Z" />
//     </svg>
//   </div>
// </div>
// <div className="info">
//   <div className="title">
//     <div className="title-container"> Overlord III </div>
//     <div className="detail-container"> Ep. 11 </div>
//   </div>
// </div>
// </div>
// <div className="anime-small-thumbnail small">
// <div className="cover">
//   <img src="https://cdn.masterani.me/wallpaper/0/1176b7Sk5J93.jpg" alt="thumbnail" className="anime-small-thumbnail" />
//   <div className="overlay hover_hidden darken">
//     <svg className="play" viewBox="0 0 24 24">
//       <path d="M8,5.14V19.14L19,12.14L8,5.14Z" />
//     </svg>
//   </div>
// </div>
// <div className="info">
//   <div className="title">
//     <div className="title-container"> Haikyuu!! Second Season </div>
//     <div className="detail-container"> Ep. 8 </div>
//   </div>
// </div>
// </div>
// <div className="anime-small-thumbnail small">
// <div className="cover">
//   <img src="https://cdn.masterani.me/wallpaper/0/226eH0orQ53.jpg" alt="thumbnail" className="anime-small-thumbnail" />
//   <div className="overlay hover_hidden darken">
//     <svg className="play" viewBox="0 0 24 24">
//       <path d="M8,5.14V19.14L19,12.14L8,5.14Z" />
//     </svg>
//   </div>
// </div>
// <div className="info">
//   <div className="title">
//     <div className="title-container"> Fullmetal Alchemist: Brotherhood </div>
//     <div className="detail-container"> Ep. 32 </div>
//   </div>
// </div>
// </div>
