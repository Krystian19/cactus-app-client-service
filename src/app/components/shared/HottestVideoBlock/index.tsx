import React from 'react';
import { RouteComponentProps } from "react-router";
import {
  Link,
  withRouter,
} from 'react-router-dom';
import { FormattedMessage } from 'react-intl';

import Episode from '../../@types/Episode';
import LazyImage from '../LazyImage';

import base64Content from '../../../utils/base64Content';

type PropType = RouteComponentProps<{}> & {
  episodes: Array<Episode>,
  title: String,
  viewAllLink?: String,
};

class HottestVideoBlock extends React.Component<PropType> {
  render() {
    const {
      episodes,
      title,
      history,
      viewAllLink,
    } = this.props;

    return (
      <div className="video-block">
        <div className="video-block-title">
          <div className="video-block-title-head">
            <span>{title}</span>
          </div>
          {viewAllLink && (
            <Link to={String(viewAllLink)} className="button">
              <FormattedMessage
                id="cactus.view_all"
                defaultMessage={'View all'}
              />
            </Link>
          )}
        </div>
        <div className="video-block-content">
          <div className="anime-small-thumbnail-list">
            {episodes.map((episode, index) => (
              <div
                key={Number(episode.id)}
                className={`anime-small-thumbnail ${(index === 0) ? 'big' : 'small'}`}
                onClick={() => history.push(`/anime/video/${Number(episode.id)}`)}
                onKeyPress={() => history.push(`/anime/video/${Number(episode.id)}`)}
                role="menuitem"
              >
                <div className="cover shimmer-load">
                  <LazyImage
                    src={`/img_cdn/${episode.Release.background}`}
                    errorSrc={base64Content.cactus_thumbnail_placeholder}
                    alt="thumbnail"
                    className="anime-small-thumbnail fade-in"
                    noLoadingPlaceholder={true}
                  />
                  <div
                    className="overlay hover_hidden darken"
                  >
                    <svg className="play" viewBox="0 0 24 24">
                      <path d="M8,5.14V19.14L19,12.14L8,5.14Z" />
                    </svg>
                  </div>
                </div>
                <div className="info">
                  <div className="title">
                    <div
                      className="title-container"
                    >
                      {`${
                        episode.Release.title
                        } (Season ${
                        episode.Release.releaseOrder
                        })`}
                    </div>
                    <div className="detail-container">
                      {`Ep. ${episode.episode_order}`}
                    </div>
                  </div>
                </div>
              </div>
            ))}
          </div>
        </div>
      </div>
    );
  }
}

export default withRouter(HottestVideoBlock);
