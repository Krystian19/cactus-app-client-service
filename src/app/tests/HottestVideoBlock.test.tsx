import React from 'react';
import { render, shallow } from 'enzyme';
import { MemoryRouter } from 'react-router-dom';

import HottestVideoBlock from '../components/shared/HottestVideoBlock';
import Episode from '../components/@types/Episode';

const episodes: Array<Episode> = [
  {
    'id': 21,
    'thumbnail': 'test.jpg',
    'episodeOrder': 12,
    'Season': {
      'id': 7,
      'seasonOrder': 1,
      'title': 'Goblin Slayer',
      'background': 'test.jpg',
      'Anime': {
        'id': 9,
        'title': 'Goblin Slayer'
      },
    },
  },
];

const componentSetup = (
  <HottestVideoBlock
    title="ðŸ”¥ right now"
    episodes={episodes}
    viewAllLink="/hottest_episodes"
  />
);

test('HottestVideoBlock component is rendering properly', () => {
  const component = render(
    <MemoryRouter>
      {componentSetup}
    </MemoryRouter>
  );

  expect(component).toMatchSnapshot();
});

test('HottestVideoBlock component shell working properly', () => {
  const component = shallow(
    <MemoryRouter>
      {componentSetup}
    </MemoryRouter>
  );

  expect(component).toMatchSnapshot();
});
