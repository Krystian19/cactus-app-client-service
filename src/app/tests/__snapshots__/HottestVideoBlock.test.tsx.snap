// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HottestVideoBlock component is rendering properly 1`] = `
<div
  class="video-block"
>
  <div
    class="video-block-title"
  >
    <div
      class="video-block-title-head"
    >
      <span>
        ðŸ”¥ right now
      </span>
    </div>
    <a
      class="button"
      href="/hottest_episodes"
    >
      <span>
        View all
      </span>
    </a>
  </div>
  <div
    class="video-block-content"
  >
    <div
      class="anime-small-thumbnail-list"
    />
  </div>
</div>
`;
